File: 1768-merge-strings-alternately/1768-merge-strings-alternately.cpp
class Solution {
public:
    string mergeAlternately(string word1, string word2) {
        string answer;
        int longestWord = std::max(word1.size(), word2.size());
        
        for (int i = 0; i < longestWord; i++) {
            if (i < word1.size())
                answer.push_back(word1[i]);
            if (i < word2.size())
                answer.push_back(word2[i]);
        }
        
        return answer;
    }
};

File: 1768-merge-strings-alternately/1768-merge-strings-alternately.rs
use std::cmp;

impl Solution {
    pub fn merge_alternately(word1: String, word2: String) -> String {
        let mut answer = String::new();
        let longest_word = cmp::max(word1.len(), word2.len());
        
        for i in 0..longest_word {
            if i < word1.len() { answer.push(word1.as_bytes()[i] as char); }
            if i < word2.len() { answer.push(word2.as_bytes()[i] as char); }
        }
        
        answer
    }
}